sophiread/FastSophiread/src/disk_io.cpp

```diff
@@ -435,3 +428,241 @@ void appendNeutronToHDF5(const std::string &out_file_name,
                          const std::vector<Neutron> &neutrons) {
   appendNeutronToHDF5(out_file_name, neutrons.cbegin(), neutrons.cend());
 }
+
+/**
+ * @brief Construct a new TPX3FileReader::TPX3FileReader object
+ *
+ * @param[in] filename
+ */
+TPX3FileReader::TPX3FileReader(const std::string &filename) {
+  fd = open(filename.c_str(), O_RDONLY);
+  if (fd == -1) {
```

Comment:
Normally this is written as
```suggestion
  if (!fd) {
```